parser/model/model.go

```diff
@@ -1160,6 +1160,9 @@ type PartitionInfo struct {
 	AddingDefinitions []PartitionDefinition `json:"adding_definitions"`
 	// DroppingDefinitions is filled when dropping a partition that is in the mid state.
 	DroppingDefinitions []PartitionDefinition `json:"dropping_definitions"`
+	// OriginalDefinitions is filled when truncating a partition that is in the mid state.
+	// original definitions used to ensure no record inserted into truncating partitions in reorg state.
+	OriginalDefinitions []PartitionDefinition `json:"original_definitions"`
```

Comment:
Why is this needed? I.e. are the full PartitionDefinition's needed or just ids?

Reply:
here you can review the new PR.